vm-setup/roles/v1aX_integration_test/tasks/upgrade.yml

```diff
@@ -0,0 +1,466 @@
+---
+  - name: Define number of BMH's
+    set_fact:
+      NUMBER_OF_BMH: "{{ NUM_OF_MASTER_REPLICAS|int +  NUM_OF_WORKER_REPLICAS|int }}"
+
+  - name: Update maxSurge and maxUnavailable fields
+    shell: |
+           kubectl get machinedeployment -n {{NAMESPACE}} test1 -o json | 
+           jq '.spec.strategy.rollingUpdate.maxSurge=1|.spec.strategy.rollingUpdate.maxUnavailable=1' | 
+           kubectl apply -f-
+    environment:
+      KUBECONFIG: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
+
+  - name: Untaint all nodes
+    shell: |
+        kubectl taint nodes --all node-role.kubernetes.io/master-
+    environment:
+      KUBECONFIG: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
+    ignore_errors: yes
+
+  - name: Scale worker down to 0
+    shell: |
+        kubectl scale machinedeployment "{{ CLUSTER_NAME }}"  -n "{{ NAMESPACE }}" --replicas=0
+    environment:
+      KUBECONFIG: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
+
+  - name: Wait until worker is scaled down and one bmh is Ready
+    shell: kubectl get bmh -n "{{ NAMESPACE }}" | grep -w ready | wc -l
+    environment:
+      KUBECONFIG: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
+    retries: 200
+    delay: 20
+    register: ready_hosts
+    until: ready_hosts.stdout == "1"
+
+# ---------- ---------- ---------- ---------- ---------- ---------- ---------- ----------
+#                       source cluster: Upgrade controlplane components                 |
+# ---------- ---------- ---------- ---------- ---------- ---------- ---------- ----------
+  - name: Gather variables that were missing on the source cluster
```

Comment:
source cluster?

Reply:
Before pivoting: Ironic runs as a docker container
After pivoting: Ironic runs as a pod. 
Source = before pivoting.

Reply:
you are right. 'source cluster' is no longer relevant.